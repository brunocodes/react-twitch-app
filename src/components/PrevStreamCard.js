import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import "./PrevStreamCard.scss";
import PropTypes from "prop-types";

const numberWithCommas = num =>
  num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");

const PrevStreamCard = ({
  isLive,
  streamLength,
  liveViewers,
  startedAt,
  url,
  imgURL,
  title,
  gameName,
  routeParam,
  views,
  streamRecordTime
}) => (
  <div className="prev-card">
    {isLive ? <p className="liveTag">LIVE</p> : ""}
    {streamLength ? (
      <p className="prev-durationTime">
        {new Date(streamLength * 1000).toISOString().substr(11, 8)}
      </p>
    ) : liveViewers ? (
      <p className="prev-viewers">
        <i className="material-icons">person</i>
        {numberWithCommas(liveViewers)}
      </p>
    ) : (
      0
    )}
    <a href={url} target="_blank" rel="noopener noreferrer">
      <img src={imgURL} alt="streamImage" />
    </a>
    <div className="prev-infoPart">
      <p className="prev-status" title={title}>
        {title}
      </p>
      <Link to={routeParam}>
        <p className="prev-gameName">{gameName}</p>
      </Link>
      {views ? (
        <p className="prev-views">{numberWithCommas(views)} views</p>
      ) : (
        <p className="prev-startedAt">{startedAt}</p>
      )}
      {streamRecordTime ? (
        <p className="prev-recordedAt">
          {moment(streamRecordTime).from(moment())}
        </p>
      ) : (
        ""
      )}
    </div>
  </div>
);

PrevStreamCard.propTypes = {
  isLive: PropTypes.bool,
  liveViewers: PropTypes.number,
  streamLength: PropTypes.number,
  views: PropTypes.number,
  url: PropTypes.string,
  imgURL: PropTypes.string,
  routeParam: PropTypes.string,
  logoURL: PropTypes.string,
  title: PropTypes.string,
  name: PropTypes.string,
  gameName: PropTypes.string,
  startedAt: PropTypes.string,
  streamRecordTime: PropTypes.string
};

export default PrevStreamCard;
