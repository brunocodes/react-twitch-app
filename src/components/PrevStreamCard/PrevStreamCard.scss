.prev {
    &-card {
      position: relative;
      background: rgb(222, 185, 253);
      float: left;
      margin: 5px;
      border-radius: 5px;
      -webkit-box-shadow: 0px 8px 17px -8px rgba(0, 0, 0, 0.75);
      -moz-box-shadow: 0px 8px 17px -8px rgba(0, 0, 0, 0.75);
      // box-shadow: 0px 8px 17px -8px rgba(0, 0, 0, 0.75);
  
      img {
        width: 320px;
        height: 180px;
        border-radius: 5px;
      }
    }
    &-durationTime {
      font-weight: bold;
      font-family: Roboto;
      position: absolute;
      margin-top: 0;
      right: 5px;
      bottom: 53px;
      color: black;
      padding-right: 3px;
      padding-left: 3px;
      font-size: 14px;
      background-color: rgba(255, 255, 255, 0.9);
      border-radius: 2px;
      display: inline-flex;
      vertical-align: middle;
      line-height: 1.5;
    }
    &-viewers {
      font-family: Roboto;
      position: absolute;
      margin-top: 0;
      bottom: 54px;
      color: red;
      padding-right: 3px;
      right: 5px;
      font-size: 14px;
      font-weight: 600;
      background-color: white;
      border-radius: 5px;
      display: inline-flex;
      vertical-align: middle;
      line-height: 1.8;
    }
    &-infoPart {
      font-family: Roboto;
      position: relative;
      top: 0px;
      left: 5px;
    }
    &-status {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      margin-top: 0;
      margin-bottom: 0px;
      font-size: 14px;
      max-width: 315px;
    }
    &-gameName {
      margin-top: 0;
      margin-bottom: 0;
      font-size: 13px;
      color: black;
    }
    &-views,
    &-startedAt {
      float: left;
      margin-top: 0;
      margin-bottom: 0;
      font-size: 13px;
    }
    &-recordedAt {
      position: relative;
      right: 8px;
      float: right;
      margin-top: 0;
      margin-bottom: 0;
      font-size: 13px;
    }
  }
  
  .liveTag {
    font-family: Roboto;
    font-weight: 900;
    position: absolute;
    margin-top: 0;
    top: 4px;
    color: red;
    padding: 0 1px;
    left: 5px;
    font-size: 14px;
    background-color: white;
    border-radius: 2px;
    display: inline-flex;
    vertical-align: middle;
    line-height: 1.5;
  }
  