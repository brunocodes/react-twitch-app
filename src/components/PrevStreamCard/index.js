import PrevStreamCard from "./PrevStreamCard";

export default PrevStreamCard;