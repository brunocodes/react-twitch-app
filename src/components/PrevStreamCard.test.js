import PrevStreamCard from "./PrevStreamCard";
import React from "react";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const wrapper = shallow(
    <PrevStreamCard
      streamLength={47319}
      streamRecordTime="2019-02-20T23:23:38Z"
      views={77705}
      gameName="Apex Legends"
      imgURL="https://static-cdn.jtvnw.net/previews-ttv/live_user_shroud-320x180.jpg"
      isLive={true}
      liveViewers={116503}
      routeParam="/game/Apex Legends"
      startedAt="started 3 hours ago"
      title="APEX NOW  @shroud on socials for updates"
      url="https=//www.twitch.tv/shroud"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
