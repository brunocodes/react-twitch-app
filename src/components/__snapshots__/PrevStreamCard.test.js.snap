// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="prev-card"
>
  <p
    className="liveTag"
  >
    LIVE
  </p>
  <p
    className="prev-durationTime"
  >
    13:08:39
  </p>
  <a
    href="https=//www.twitch.tv/shroud"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="streamImage"
      src="https://static-cdn.jtvnw.net/previews-ttv/live_user_shroud-320x180.jpg"
    />
  </a>
  <div
    className="prev-infoPart"
  >
    <p
      className="prev-status"
      title="APEX NOW  @shroud on socials for updates"
    >
      APEX NOW  @shroud on socials for updates
    </p>
    <Link
      replace={false}
      to="/game/Apex Legends"
    >
      <p
        className="prev-gameName"
      >
        Apex Legends
      </p>
    </Link>
    <p
      className="prev-views"
    >
      77,705
       views
    </p>
    <p
      className="prev-recordedAt"
    >
      14 days ago
    </p>
  </div>
</div>
`;
